Automattic/woocommerce-payments
# Add Multi-Currency Support to Page Caching via Cookies

Fixes #8130 

#### Changes proposed in this Pull Request

This PR introduces the `wcpay_currency` cookie, which consolidates the selected currency code and exchange rate into a single cookie, intended primarily for caching purposes on the frontend. This enhancement allows merchants to use this cookie within reverse proxy setups to efficiently cache product pages based on the currency. 

Note: The WooCommerce session cookie must be included in requests to accurately set currency cookies. If testing a reverse proxy solution involving these cookies, ensure caching rules do not strip the WooCommerce session cookies to preserve the multi-currency functionality.

#### Testing instructions

**Test 1: Multi-Currency Feature Disabled**

1. Ensure the multi-currency feature is disabled in WooPayments settings.
2. Clear any existing cookies related to your site to start with a clean slate.
3. Navigate through your site, visiting various pages including product pages, the homepage, and any other relevant pages.
4. Check your browser's developer tools under the Application or Storage tab to verify that the `wcpay_currency` cookie is not set.
5. Confirm that navigating through the site does not trigger these cookies to be set, indicating that with the multi-currency feature turned off, the mechanism is inactive.

**Test 2: Multi-Currency Feature Enabled with Currency Switching**

1. Enable the multi-currency feature in WooPayments settings and configure a few different currencies.
2. Clear your browser's cookies to ensure a fresh start for this test.
3. Visit your site and switch to a different currency using the currency switcher provided by WooPayments.
4. After switching currencies, check your browser's developer tools under the Application or Storage tab for the `wcpay_currency` cookie.
5. Verify that the `wcpay_currency` cookie reflects the code of the newly selected currency.
6. Continue to switch between different currencies and repeat steps 4-6, confirming that each time the currency is switched, both cookies update accurately to reflect the new currency code and exchange rate.

-------------------


**Post merge**
